### DISCUSSION TITLE
Ms sql service with full text search

### DISCUSSION THREAD

**Discussion by krasavindy** [community/community#25464] [Actions]
<p>we using ms sql server as service in our workflow. Now we need to use full text search on it, but fts not installed in official sql server dicker images, and requires additional installation. Is it possible to install it?</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/krasavindy">@krasavindy</a>,</p>
<p>To install tools in a container, normally we have the two ways below:</p>
<ol>
<li>
<p>Add the following commands (see <a href="https://docs.microsoft.com/en-us/sql/linux/sql-server-linux-setup-full-text-search?view=sql-server-ver15#ubuntu" rel="noopener nofollow ugc"><strong>Install the mssql-server-fts on Ubuntu</strong></a>) to the <strong>RUN</strong> instruction in the <strong>Dockerfile</strong>. When building the image, the <strong>mssql-server-fts</strong> will be installed.</p>
<pre><code class="lang-bash">sudo apt-get update 
sudo apt-get install -y mssql-server-fts
</code></pre>
<p>In this way, you need to create a customized docker image based on the official sql server docker image, and add above commands in the <strong>Dockerfile</strong> of the customized image. Then you can use this customized image in your workflow.</p>
</li>
<li>
<p>If you do not want to create new image, after the container started, you can execute above commands in the container to install the <strong>mssql-server-fts</strong>.<br>
Unlike container job (<a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idcontainer" rel="noopener nofollow ugc"><strong>jobs.&lt;job_id&gt;.container</strong></a>), the steps in the job will not run the service container. However, there is an option to set volumes for the service container to use (<a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idservicesservice_idvolumes" rel="noopener nofollow ugc"><strong>jobs.&lt;job_id&gt;.services.&lt;service_id&gt;.volumes</strong></a>).<br>
You can try to use above commands to install the <strong>mssql-server-fts</strong> on the runner machine, then set a volume to share the installation directory of <strong>mssql-server-fts</strong> between runner machine and service container.</p>
</li>
</ol>
